src/dataset.py

```diff
@@ -0,0 +1,120 @@
+from pathlib import Path
+from typing import List, Tuple
+
+import torch
+from omegaconf import OmegaConf
+from tokenizers import SentencePieceBPETokenizer
+from torch.utils.data import Dataset
+
+root_dir = Path(__file__).parent.parent
+model_config_dir = root_dir / "configs" / "model"
+tokenizer_config_dir = root_dir / "configs" / "tokenizer"
+tokenizer_dir = root_dir / "tokenizer"
+
+
+class WMT14Dataset(Dataset):
+    """Dataset to train and test transformer
+
+    Attributes:
+        langpair: language pair to translate
+        source_lines: list of text in source language
+        target_lines: list of text in target language
+    """
+
+    def __init__(
+        self, langpair: str, source_lines: List[str], target_lines: List[str]
+    ) -> None:
+        super().__init__()
+        if langpair in ["de-en", "en-de", "deen", "ende"]:
+            self.tokenizer_config_path = (
+                tokenizer_config_dir / "sentencepiece_bpe_wmt14_deen.yaml"
+            )
```

Comment:
이거는 suggenstion은 아니구요.
저는 보통 이런 방식도 즐겨 사용하긴합니다만, 요런건 어떻게 생각하시나요?!

```suggestion
class WMT14Dataset(Dataset):
    """Dataset to train and test transformer

    Attributes:
        langpair: language pair to translate
        source_lines: list of text in source language
        target_lines: list of text in target language
    """
    LANG: List[str] = ["de-en", "en-de", "deen", "ende"]
    def __init__(
        self, langpair: str, source_lines: List[str], target_lines: List[str]
    ) -> None:
        super().__init__()
        if langpair in self.LANG:
            self.tokenizer_config_path = (
                tokenizer_config_dir / "sentencepiece_bpe_wmt14_deen.yaml"
            )
```

Reply:
앗 근데 이런 방식은 다른 언어쌍이 들어올 때 대응이 안되지 않나요..? 
여기서는 보통 번역 쪽에서 langpair 입력할 때 다양한 방식이 있는데 그 처리를 해주고 싶었어요 ㅋㅋㅋ 

registry 도 한번 볼게요!

Reply:
올려주신 detectron dataset 은, 정말 데이터와 metadata를 등록해두고 쓰는 것 같은데 여기에는 적합한지 잘 모르겠어요 ㅋㅋ
여기서 체크하는 것은, 학습시에 언어쌍을 지정해서 사용할텐데요, 그 때 불러올 데이터셋과 토크나이저가 지정된 언어쌍에 대해서 준비된 것인지 확인하는 용도라서요! ㅎㅎ

Reply:
앗앗... 제가 NLP는 알못이라서ㅎㅎ
그냥 이런 것도 있다? 이런 느낌으로 던져보긴 했습니다. 🙇‍♂️
보섭님이 피드백 주시면 좋을 것 같긴한데.... @aisolab 호출!